lib/personalized_search/available_attribute_groups.dart

```diff
@@ -0,0 +1,53 @@
+import 'dart:convert';
+import 'package:openfoodfacts/model/AttributeGroup.dart';
+
+/// Referential of attribute groups, with loader.
+class AvailableAttributeGroups {
+  /// Load constructor; may throw an exception.
+  AvailableAttributeGroups.loadFromJSONString(
+    final String attributeGroupsString,
+  ) {
+    final dynamic inputJson = json.decode(attributeGroupsString);
+    final List<AttributeGroup> attributeGroups = <AttributeGroup>[];
+    for (final dynamic item in inputJson as List<dynamic>) {
+      attributeGroups.add(AttributeGroup.fromJson(item));
+    }
+    if (attributeGroups.isEmpty) {
+      throw Exception(
+          'Unexpected error: empty attribute groups from json string $attributeGroupsString');
+    }
+    _attributeGroups = attributeGroups;
+  }
+
+  List<AttributeGroup>? _attributeGroups;
+
+  List<AttributeGroup>? get attributeGroups => _attributeGroups;
+
+  /// Where a localized JSON file can be found.
+  /// [languageCode] is a 2-letter language code.
+  static String getUrl(final String languageCode) =>
+      'https://world.openfoodfacts.org/api/v2/attribute_groups?lc=$languageCode';
+
+  static const String ATTRIBUTE_GROUP_INGREDIENT_ANALYSIS =
+      'ingredients_analysis';
+  static const String ATTRIBUTE_GROUP_NUTRITIONAL_QUALITY =
+      'nutritional_quality';
+  static const String ATTRIBUTE_GROUP_PROCESSING = 'processing';
+  static const String ATTRIBUTE_GROUP_ALLERGENS = 'allergens';
+  static const String ATTRIBUTE_GROUP_LABELS = 'labels';
+  static const String ATTRIBUTE_GROUP_ENVIRONMENT = 'environment';
+
+//  static const String ATTRIBUTE_VEGETARIAN = 'vegetarian';
```

Comment:
Minor: there is a bit of commented out code. I'd just delete it 😊

Reply:
@peterwvj Fair enough. The following question is: would that be relevant instead to put a maximum of attribute ids (all of them?), rather than just a limited set?
I know that in smoothie I had to add or use some of those ids, and there's a good probability that someone who cares about sesame seeds or fair trade would be glad to find the ids already there as `const String`s.

Reply:
Well, if they are helpful why not just comment them back in? 😊 My point is that commented out code rot.

Reply:
Sure, putting there the complete list would make sense - even uncommenting those lines wouldn't be comprehensive.

Reply:
I uncommented and completed the strings (allergens etc.) and moved them to the Attribute and AttributeGroup classes.

Reply:
Thanks. I have no further comments 👍